crowbar_framework/app/models/neutron_service.rb

```diff
@@ -94,14 +94,31 @@ def validate_proposal_after_save proposal
     end
 
     plugin = proposal["attributes"]["neutron"]["networking_plugin"]
-    mode = proposal["attributes"]["neutron"]["networking_mode"]
+    ml2_mechanism_drivers = proposal["attributes"]["neutron"]["ml2_mechanism_drivers"]
+    ml2_type_drivers = proposal["attributes"]["neutron"]["ml2_type_drivers"]
 
-    if %(linuxbridge cisco).include?(plugin) and mode != "vlan"
-      validation_error("The \"#{plugin}\" plugin only supports the mode: \"vlan\"")
+    # validate given comma seperated driver list
+    ml2_mechanism_drivers_valid = %w(linuxbridge openvswitch cisco_nexus")
+    ml2_type_drivers_valid = %w(gre vlan")
+
+    ml2_mechanism_drivers_proposed = ml2_mechanism_drivers.split(",")
+    ml2_mechanism_drivers_proposed.each do |drv|
+      unless ml2_mechanism_drivers_valid.include? drv.strip
+        validation_error("ML2 mechanism driver '#{drv}' not allowed. Valid drivers are: #{ml2_mechanism_drivers_valid.join(', ')}")
+      end
+    end
+
+    ml2_type_drivers_proposed = ml2_type_drivers.split(",")
+    ml2_type_drivers_proposed.each do |drv|
+      unless ml2_type_drivers_valid.include? drv.strip
+        validation_error("ML2 type driver '#{drv}' not allowed. Valid drivers are: #{ml2_type_drivers_valid.join(', ')}")
+      end
     end
 
-    if !%(gre vlan).include?(mode)
-      validation_error("Unknown networking mode \"#{mode}\"")
+    %w(linuxbridge cisco_nexus).each do |type_drv|
```

Comment:
suggest to rename |type_drv| to |drv| since its a mechanism driver, not a type driver